server/graphql/v2/mutation/CommentReactionMutations.js

```diff
@@ -0,0 +1,56 @@
+import { GraphQLNonNull, GraphQLString } from 'graphql';
+import { CommentReaction } from '../object/CommentReaction';
+import models from '../../../models';
+import { FeatureNotAllowedForUser, Unauthorized } from '../../errors';
+import { canUseFeature } from '../../../lib/user-permissions';
+import FEATURE from '../../../constants/feature';
+import { CommentReactionCreateInput } from '../input/CommentReactionCreateInput';
+
+async function createCommentReaction(entity, args, req) {
+  if (!req.remoteUser) {
+    throw new Unauthorized();
+  } else if (!canUseFeature(req.remoteUser, FEATURE.CONVERSATIONS)) {
+    throw new FeatureNotAllowedForUser();
+  }
+
+  const { emoji, comment } = args;
+
+  return await models.CommentReaction.addCommentReaction(req.remoteUser, comment, emoji);
+}
+
+async function deleteCommentReaction(id, remoteUser) {
+  if (!remoteUser) {
+    throw new Unauthorized();
+  } else if (!canUseFeature(remoteUser, FEATURE.CONVERSATIONS)) {
+    throw new FeatureNotAllowedForUser();
+  }
+  return await models.CommentReaction.addCommentReaction(id);
```

Comment:
We need to make sure that user is allowed to delete the reaction, aka `LoggedInUser.isAdmin(reaction.FromCollectiveId)`

Reply:
@Betree : So I've added the condition such that the user who owns the comment reaction to be able to delete it. But the statement `LoggedInUser.isAdmin(reaction.FromCollectiveId)` implies that the admin of the collective should also be able to delete the comment reaction? Is this something we should allow; just wondering since for example GitHub doesn't allow that. Only the creator of the reaction can delete the reaction right? :thinking:

Reply:
`reaction.FromCollectiveId` will reference user's collective (`LoggedInUser.collective.id`). So users will only be able to delete their own reactions.

Reply:
We tend to rely on the `CollectiveId` to manage permissions. `UserId` are mostly used for logging/history purposes. The idea is that if you post a comment on behalf of your collective/organization, all admins of this profile should be able to edit/delete the comment.

Reply:
@Betree : Wonderful. Thanks much for the information. :smile: